### PR TITLE
new upload procedure

### PR DESCRIPTION
**Pull request by pkasprzyk**

### CODE REVIEW

File: neptune/internal/storage/datastream.py [hunk 1]
```diff
@@ -16,29 +16,65 @@
...
+class FileChunker:
+    def __init__(self, filename, fobj, total_size, multipart_config: MultipartConfig):
+        self._filename = filename
+        self._fobj = fobj
+        self._total_size = total_size
+        self._min_chunk_size = multipart_config.min_chunk_size
+        self._max_chunk_size = multipart_config.max_chunk_size
+        self._max_chunk_count = multipart_config.max_chunk_count
+
+    def _get_chunk_size(self) -> int:
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/internal/storage/datastream.py:47]: Is it, actually? That's just my implementation decision ;)

File: neptune/internal/storage/datastream.py [hunk 2]
```diff
@@ -16,29 +16,65 @@
...
+    def generate(self) -> Generator[FileChunk, Any, None]:
+        chunk_size = self._get_chunk_size()
+        last_offset = 0
+        while last_offset < self._total_size:
+            chunk = self._fobj.read(chunk_size)
+            if chunk:
+                if isinstance(chunk, str):
+                    chunk = chunk.encode("utf-8")
+                new_offset = last_offset + len(chunk)
+                yield FileChunk(chunk, last_offset, new_offset)
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/internal/storage/datastream.py]: done

File: neptune/internal/storage/datastream.py [hunk 3]
```diff
@@ -16,29 +16,65 @@
...
 
 from neptune.internal.storage.storage_utils import (
     UploadEntry,
     AttributeUploadConfiguration,
 )
+from neptune.new.exceptions import InternalClientError
+from neptune.new.internal.backends.api_model import MultipartConfig
 
 
 class FileChunk(object):
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/internal/storage/datastream.py]: not really my code, but you're right :)

[the remark this answers is not in the document]
Comment (shnela) [reply neptune/internal/storage/datastream.py]: You touched this, it's on you :p

File: neptune/internal/storage/datastream.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shnela) [reply neptune/internal/storage/datastream.py:47]: Ok

**Review by shnela — DISMISSED** [review]

File: neptune/new/internal/backends/hosted_file_operations.py [hunk 4]
```diff
@@ -56,55 +60,62 @@
...
+
 def upload_file_attribute(
     swagger_client: SwaggerClient,
     container_id: str,
     attribute: str,
     source: Union[str, bytes],
     ext: str,
-) -> Optional[NeptuneException]:
+    multipart_config: Optional[MultipartConfig],
+) -> List[NeptuneException]:
```
[the remark this answers is not in the document]
Comment (pkasprzyk) [reply neptune/new/internal/backends/hosted_file_operations.py:75]: exactly that, I wanted consistency

**Review by shnela — APPROVED** [review]

**Pull request merged**